### PR TITLE
Trigger e2etests for PR in st2 repo

### PR DESCRIPTION
**Pull request by armab**
Related: 
- https://github.com/StackStorm/st2/pull/3625
- https://github.com/StackStorm/st2cd/pull/295

There are a lot of changes in `st2` with CircleCI2.0 + TravisCI overhaul.
- Building the package in `st2` repo is triggered natively by CircleCI itself now (no rules needed)
- We're using CircleCI 2.0 with Workflows (several sub-tasks might complicate everything)
- Artifacts dirs were changed
- We build only `st2` package in `st2` repo (without `st2mistral`)
  - Requires `curl|bash` installer changes

Because of the significantly changed logic, this turns into a bigger refactoring existing `st2ci` workflows.

### CODE REVIEW

File: actions/st2_pkg_e2e_test.meta.yaml [hunk 1]
```diff
@@ -51,8 +51,8 @@ parameters:
     version:
         type: string
     dev_build:
-        type: integer
-        description: CircleCI build number to install artifacts from. Overrides pkg_env, release and version.
+        type: string
+        description: Repo name + CircleCI build number ('st2/3126') to install artifacts from. Overrides pkg_env, release and version.
```
Comment (armab) [inline actions/st2_pkg_e2e_test.meta.yaml]: See https://github.com/StackStorm/st2-packages/pull/476 why

**Comment by Kami** [comment]
LGTM from a high level perspective.

I assume you will test it and ensure it works end to end so we should be fine.

**Comment by armab** [comment]
@humblearner as said before, I'm pulling only these specific file changes, to make sure pack-reinstall won't break any of your WIP changes.

**Comment by humblearner** [comment]
Thanks @armab, I have the changes here: https://github.com/StackStorm/st2cd/pull/298 and https://github.com/StackStorm/st2ci/pull/104/files

**Pull request merged**